crates/core/src/models/transaction.rs

```diff
@@ -72,15 +72,13 @@ impl ConvertibleStarknetTransaction for StarknetTransaction {
         block_number: Option<U256>,
         transaction_index: Option<U256>,
     ) -> Result<EthTransaction, EthApiError> {
-        let starknet_block_latest = StarknetBlockId::Tag(BlockTag::Latest);
-        let sender_address: FieldElement = self.sender_address()?.into();
-
-        let class_hash = client.inner().get_class_hash_at(starknet_block_latest, sender_address).await?;
-
-        if class_hash != client.proxy_account_class_hash() {
+        if !self.is_kakarot_tx(client).await? {
```

Comment:
early returns remind me of TS, I like:)
can we check if this is also idiomatic in Rust?